### ISSUE TITLE
Grid getRowBy<X> and rows not currently in the virtual container

### ISSUE THREAD

**Issue by damyanpetev** [IgniteUI/igniteui-angular#6158]
Currently the Grid API for rows like [`getRowByKey`](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxgridcomponent.html#getrowbykey) (assuming `getRowByIndex` as well) operate with actual `IgxRowComponent` instances and won't return a result, unless the row is actually in the current virtualization chunk.

There a few issues with that:
- Nowhere in the description does it say these will return undefined based on chunk
- It's a public API, in fact the only API for getting rows, and it's only useful for a limited number of scenarios using interaction events (that probably provide row in the args anyway)
- Since that's the only API, getting something like the data view index (need for `navigateTo`) can only be done by searching the `dataView`. That's fine, but again, should at least be documented

If I remember correctly, @rkaraivanov was in favor of using a row-like interface for API rather than the actual component instance exactly because those are not available all the time, so we might be in for a larger discussion here, which is why this is not quite marked as a documentation/bug/feature.

**Comment by julianobrasil** [comment]
For me, this is really a must-have feature. I've spent an entire day yesterday and I found this today (thanks, @damyanpetev, I owe you a beer). Documenting dataView usage (and maybe its interaction with navigateTo) would be great here! 

My use case: I have a 700-rows data in a grid with like 12 rows rendered at a given time. The data is multilevel and I'm using `IgxTreeGridComponent`. In this specific case, if I add a new editable row at the first level, the newly added row will be at the end of the grid. In my case, the first level was the parent of the remaining 699 rows, so the new added row is created completely out of the user's visible area of the grid. I'd like to navigate to the newly added row, at the end of the grid, get a reference for it and programmatically start the edition in the input of a specific column. `navigateTo` does most of the job, but without knowing the row index of the row, it's useless.
